src/ResourceManager/Common/Commands.Common.Strategies/ResourceId.cs

```diff
@@ -0,0 +1,79 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Common.Strategies
+{
+    public static class ResourceId
+    {
+        public const string Subscriptions = "subscriptions";
+        public const string ResourceGroups = "resourceGroups";
+        public const string Providers = "providers";
+
+        /// <summary>
+        /// Returns 'null' if the given id is not parsable.
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        public static IResourceId TryParse(string id)
```

Comment:
@sergey-shandar any reason you implemented this class instead of using the existing [`ResourceIdentifier`](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Common/Commands.ResourceManager.Common/Utilities/Models/ResourceIdentifier.cs) class?

Reply:
We also need this interface in template implementation. The ARM template implementation is  inside a Strategy library which doesn't depend on a library which implements `ResourceIdentifier` class.